src/reactions/MessageReactionList.js

```diff
@@ -0,0 +1,98 @@
+/* @flow strict-local */
+import React, { PureComponent } from 'react';
+import { View } from 'react-native';
+import { TabNavigator, TabBarTop } from 'react-navigation';
+import type { NavigationScreenProp } from 'react-navigation';
+
+import ReactionUserList from './ReactionUserList';
+import { connect } from '../react-redux';
+import type { AggregatedReaction, Dispatch, Message, UserOrBot } from '../types';
+import { Screen, RawLabel } from '../common';
+import { getOwnUser } from '../selectors';
+import aggregateReactions from './aggregateReactions';
+import styles from '../styles';
+import { getAllUsersById } from '../users/userSelectors';
+import tabsOptions from '../styles/tabs';
+import Emoji from '../emoji/Emoji';
+import { objectFromEntries } from '../jsBackport';
+
+const getReactionsTabs = (
+  aggregatedReactions: $ReadOnlyArray<AggregatedReaction>,
+  targetReactionName: string,
+  usersById: Map<number, UserOrBot>,
+) => {
+  const reactionsTabs = objectFromEntries(
+    aggregatedReactions.map(aggregatedReaction => [
+      aggregatedReaction.name,
+      {
+        screen: () => (
+          <ReactionUserList reactedUserIds={aggregatedReaction.users} users={usersById} />
+        ),
+        navigationOptions: {
+          tabBarLabel: () => (
+            <View style={styles.row}>
+              <Emoji name={aggregatedReaction.name} />
+              <RawLabel style={styles.paddingLeft} text={`${aggregatedReaction.count}`} />
+            </View>
+          ),
+        },
+      },
+    ]),
+  );
+
+  return TabNavigator(reactionsTabs, {
+    backBehavior: 'none',
+    initialRouteName: targetReactionName,
+    tabBarComponent: TabBarTop,
+    tabBarPosition: 'top',
+    ...tabsOptions({
+      showLabel: true,
+      showIcon: false,
+      style: {
+        borderWidth: 0.15,
+      },
+    }),
+  });
+};
+
+type NavigationsProps = {|
+  navigation: NavigationScreenProp<{ params: {| targetReactionName: string, messageId: number |} }>,
```

Comment:
How about just `reactionName`? I don't think the word "target" adds much information here, or even makes a lot of sense outside the few lines in js.js that fire off the event to the outside.